server/src/main/java/com/flipkart/varadhi/events/EventManager.java

```diff
@@ -0,0 +1,150 @@
+package com.flipkart.varadhi.events;
+
+import com.flipkart.varadhi.cluster.MessageRouter;
+import com.flipkart.varadhi.cluster.VaradhiClusterManager;
+import com.flipkart.varadhi.core.cluster.entities.MemberInfo;
+import com.flipkart.varadhi.entities.auth.ResourceType;
+import com.flipkart.varadhi.produce.providers.TopicProvider;
+import com.flipkart.varadhi.providers.ProjectProvider;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Objects;
+
+import static com.flipkart.varadhi.common.Constants.ENTITY_EVENTS_HANDLER;
+
+/**
+ * Central manager for entity events and cache management in Varadhi.
+ * <p>
+ * This class is responsible for:
+ * <ul>
+ *   <li>Preloading entity caches for improved performance</li>
+ *   <li>Setting up event handlers for entity events</li>
+ *   <li>Coordinating event distribution across the system</li>
+ * </ul>
+ * <p>
+ * This component is designed to be initialized once at application startup,
+ * before any verticles are deployed.
+ *
+ * @see EntityEventDispatcher
+ * @see ProjectProvider
+ * @see TopicProvider
+ */
+@Slf4j
+public final class EventManager {
+
+    /**
+     * Provider for project entities.
+     */
+    private final ProjectProvider projectProvider;
+
+    /**
+     * Provider for topic entities.
+     */
+    private final TopicProvider topicProvider;
```

Comment:
also instead of having each "readCache" explicitly in here, does it make sense to have it like

Map<ResourceType, ReadCache<?>> so that this class can be used as a singleton to register any readCache once.